coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -67,6 +68,29 @@ public static CoralSpark create(RelNode irRelNode) {
     return new CoralSpark(baseTables, sparkUDFInfos, sparkSQL);
   }
 
+  /**
+   * Users use this function as the main API for getting CoralSpark instance.
+   * This should be used when user need to align the Coral-spark translated SQL
+   * with Coral-schema output schema
+   *
+   * @param irRelNode An IR RelNode for which CoralSpark will be constructed.
+   * @param schema Coral schema that is represented by an Avro schema
+   * @return [[CoralSparkInfo]]
+   */
+  public static CoralSpark createWithCoralSchema(RelNode irRelNode, Schema schema) {
```

Comment:
Let us use the same name as the original `create` method, and consider this as an override.

Reply:
I think you meant "overload" :)